geode-wan-txgrouping/src/distributedTest/java/org/apache/geode/internal/cache/wan/txgrouping/parallel/TxGroupingParallelDUnitTest.java

```diff
@@ -0,0 +1,506 @@
+    newYorkServerVM
+        .invoke(() -> validateRegionSize(REGION_NAME, transactions * putsPerTransaction));
+
+    List<Integer> londonServerStats =
+        getSenderStats(newYorkName, 0, (VM[]) ArrayUtils.remove(londonServersVM, 0));
+
+    // queue size
+    assertThat(londonServerStats.get(0)).isEqualTo(0);
+
+    // eventsReceived
+    // We may see two retried events (as transactions are made of 2 events) on all members due to
+    // the kill
+    assertThat(londonServerStats.get(1)).isLessThanOrEqualTo((entries + 2) * redundantCopies);
+    assertThat(londonServerStats.get(1)).isGreaterThanOrEqualTo(entries * redundantCopies);
+
+    // queuedEvents
+    assertThat(londonServerStats.get(2)).isLessThanOrEqualTo((entries + 2) * redundantCopies);
+    assertThat(londonServerStats.get(2)).isGreaterThanOrEqualTo(entries * redundantCopies);
+
+    // batches redistributed
+    assertThat(londonServerStats.get(5)).isEqualTo(0);
+
+    // batchesReceived is equal to numberOfEntries/(batchSize+1)
+    // As transactions are 2 events long, for each batch it will always be necessary to
+    // add one more entry to the 9 events batch in order to have complete transactions in the batch.
+    int batchesReceived = (entries) / (batchSize + 1);
+    newYorkServerVM.invoke(() -> checkGatewayReceiverStatsHA(batchesReceived, entries, entries));
+  }
+
+  private void checkQueuesAreEmptyAndOnlyCompleteTransactionsAreReplicated(String senderId,
+      boolean isBatchesRedistributed) {
+    List<Integer> senderStatsLondonServers = getSenderStats(senderId, 0, londonServersVM);
+
+    // queue size:
+    assertThat(senderStatsLondonServers.get(0)).isEqualTo(0);
+    // batches redistributed:
+    int batchesRedistributed = senderStatsLondonServers.get(5);
+    if (isBatchesRedistributed) {
+      assertThat(batchesRedistributed).isGreaterThan(0);
+    } else {
+      assertThat(batchesRedistributed).isEqualTo(0);
+    }
+    // batches with incomplete transactions
+    assertThat(senderStatsLondonServers.get(13)).isEqualTo(0);
+
+    for (VM londonServer : londonServersVM) {
+      londonServer.invoke(() -> validateGatewaySenderQueueAllBucketsDrained(senderId));
+    }
+  }
+
+  protected void validateGatewaySenderQueueAllBucketsDrained(final String senderId) {
+    IgnoredException exp =
+        IgnoredException.addIgnoredException(RegionDestroyedException.class.getName());
+    IgnoredException exp1 =
+        IgnoredException.addIgnoredException(ForceReattemptException.class.getName());
+    try {
+      GatewaySender sender = getGatewaySender(senderId);
+      final AbstractGatewaySender abstractSender = (AbstractGatewaySender) sender;
+      await().untilAsserted(() -> assertThat(abstractSender.getEventQueueSize()).isEqualTo(0));
+      await().untilAsserted(
+          () -> assertThat(abstractSender.getSecondaryEventQueueSize()).isEqualTo(0));
+    } finally {
+      exp.remove();
+      exp1.remove();
+    }
```

Comment:
`IgnoredException` implements `AutoCloseable` so you can use it in try-with-resources syntax. Using import-static and `addIgnoredException(Class)` tidies it up nicely:
```
import static org.apache.geode.test.dunit.IgnoredException.addIgnoredException;

try (IgnoredException ie1 = addIgnoredException(RegionDestroyedException.class);
     IgnoredException ie2 = addIgnoredException(ForceReattemptException.class)) {
  // ...
}